stress_test/controller.py

```diff
@@ -0,0 +1,402 @@
+# Copyright (c) 2016 Intracom S.A. Telecom Solutions. All rights reserved.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License v1.0 which accompanies this distribution,
+# and is available at http://www.eclipse.org/legal/epl-v10.html
+
+""" Controller Class- All controller-related functionality is here"""
+
+import json
+import common
+import logging
+import os
+import subprocess
+import time
+import util.netutil
+import util.process
+import conf_collections_util
+import requests
+import emulators.nb_generator.flow_utils
+
+
+
+class Controller:
+
+
+    def __init__(self, ctrl_base_dir, test_config):
+
+        """Create a Controller object. Options from JSON input file
+        :param test_config: JSON input configuration
+        :type test_config: JSON configuration dictionary
+        """
+        self.name = test_config['controller_name']
+        logging.info('Initializing parameters for the {0} Controller '.format(self.name))
+
+        self.base_dir = ctrl_base_dir
+        self.ip = test_config['controller_node_ip']
+        self.ssh_port = test_config['controller_node_ssh_port']
+        self.username = test_config['controller_node_username']
+        self.password = test_config['controller_password']
+
+        self.need_rebuild = self.base_dir + test_config['controller_rebuild']
+        self.need_cleanup = self.base_dir + test_config['controller_cleanup']
+        self.of_port = test_config['controller_port']
+        self.cpu_shares = test_config['controller_cpu_shares']
+        self.logs_dir = self.base_dir + test_config['controller_logs_dir']
+
+        self.build_hnd = self.base_dir + test_config['controller_build_handler']
+        self.start_hnd = self.base_dir + test_config['controller_start_handler']
+        self.stop_hnd = self.base_dir + test_config['controller_stop_handler']
+        self.status_hnd = self.base_dir + test_config['controller_status_handler']
+        self.clean_hnd = self.base_dir + test_config['controller_clean_handler']
+
+        self.status = 'UNKNOWN'
+
+        self.java_opts = ' '.join(test_config['java_opts'])
+        self.pid = -1
+
+        self._init_ssh = None
+
+        self_node = conf_collections_util.node_parameters(self.name,
+                                      self.ip,
+                                      int(self.ssh_port),
+                                      self.username,
+                                      self.password)
+
+    def init_ssh(self)
+        logging.info(
+            '[open_ssh_connection] Initiating SSH session with {0} node.'.
+            format(self.name))
+        self._init_ssh = util.netutil.ssh_connect_or_return(connection, 10)
```

Comment:
We should refactor `ssh_connect_or_return` to use directly the fields of `connection` tuple. 

We started using named tuples to attack the "too many function arguments" quality issue, but here we essentially have 5 arguments (ip, ssh_port, ssh_user, ssh_pass, 10) which could simply be passed directly. 

If we refactor _now_ `ssh_connect_or_return` to accept these 5 args (instead of 2), this will break other "clients" that call this function in its current, 2-args form. Therefore, I propose to implement a temporary clone of the function (e.g. name it `__ssh_connect_or_return`), which will be used by `init_ssh`, and will be subsequently used by all other clients as they are being refactored, as well. 

The only difference between `__ssh_connect_or_return` and `ssh_connect_or_return` will be the arguments list. 

At the end, when no one will use `ssh_connect_or_return`, we will rename `__ssh_connect_or_return` back to its normal name (`ssh_connect_or_return`)